sys/kern/src/arch/riscv32.rs

```diff
@@ -569,8 +560,22 @@ fn trap_handler(task: &mut task::Task) {
         Trap::Exception(Exception::InstructionFault) => unsafe {
             handle_fault(task, FaultInfo::IllegalText);
         },
+
         _ => {
-            panic!("Unimplemented exception");
+            if mcause.is_exception() == true {
+                panic!("Unimplemented exception");
+            } else if (mcause.cause()
+                == Trap::Interrupt(Interrupt::MachineExternal))
+                || (mcause.code() >= 16)
+            {
+                // TODO: Put the second condition under a cfg
```

Comment:
This `TODO` is pretty vague, Is this tracked in jira?

Reply:
Meant to handle that in this patch, fixed.